src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextEncoder.cs

```diff
@@ -516,23 +515,22 @@ internal void WriteEntityRef(string name)
         //
         // Private implementation methods
         //
-        // This is a helper method to workaround the fact that TextWriter does not have a Write method
-        // for fragment of a string such as Write( string, offset, count).
-        // The string fragment will be written out by copying into a small helper buffer and then
-        // calling textWriter to write out the buffer.
-        private void WriteStringFragment(string str, int offset, int count, char[] helperBuffer)
+        private void WriteStringFragment(ReadOnlySpan<char> ros)
```

Comment:
My original suggestion was to move this method altogether.

Reply:
@eiriktsarpalis @krwq is it definitely worth deleting it?

Reply:
Based on the original comments in the method, this was intended as a workaround for the lack of a TextWriter method that accepts slices.

Reply:
@eiriktsarpalis But the method accepted a string, which could also be written in one go.

Reply:
@eiriktsarpalis done